lib/src/ui/pages/dashboard/news/most_popular/most_popular_news_notifier.dart

```diff
@@ -1,92 +1,54 @@
-import 'package:allthenews/src/ui/pages/dashboard/news/secondary_news/news_paginated_view_entity.dart';
-import 'package:flutter/foundation.dart';
+import 'package:allthenews/src/ui/pages/dashboard/news/secondary_news/secondary_news_list_entity.dart';
 
-import 'most_popular_news_view_entity.dart';
-import 'most_popular_view_state.dart';
-
-abstract class _Constants {
-  static const pageSize = 10;
-}
-
-class MostPopularNewsNotifier extends ChangeNotifier {
-  final NYTimesPaginatedRepository _nyTimesPaginatedRepository;
+class MostPopularNewsNotifier extends PaginationNotifier<Article, SecondaryNewsListEntity> {
   final SettingsRepository _settingsRepository;
 
-  MostPopularViewState _state = const MostPopularViewState();
-
-  MostPopularNewsViewEntity get _loadedMostPopularViewEntity => _state.viewEntity;
+  String _mostPopularNewsPageTitle = '';
 
-  NewsPaginatedViewEntities get _loadedNewsPaginatedViewEntities => _state.viewEntity?.newsPaginatedViewEntities;
+  MostPopularPaginatedViewState<SecondaryNewsListEntity> get mostPopularViewState => MostPopularPaginatedViewState(
+        super.state,
+        _mostPopularNewsPageTitle,
+      );
 
-  MostPopularNewsNotifier(this._nyTimesPaginatedRepository, this._settingsRepository);
-
-  MostPopularViewState get state => _state;
+  MostPopularNewsNotifier(
+    GetPageUseCase<Article> getPageUseCase,
+    Mapper<Article, SecondaryNewsListEntity> mapper,
+    this._settingsRepository,
+  ) : super(getPageUseCase, mapper);
 
+  @override
   Future<void> loadFirstPage() async {
-    _setNotifierState(const MostPopularViewState(isLoading: true));
-    final Future<List<Article>> articles =
-        _nyTimesPaginatedRepository.getMostPopularArticlesPage(Page(1, _Constants.pageSize)).catchError((error) => _onFetchError(error));
-
-    final Future<String> popularNewsTitle =
-        _settingsRepository.getPopularNewsCriterion().then((criterion) => criterion.getTitle()).catchError((error) => _onFetchError(error));
-
-    _setLoadedState(await articles, await popularNewsTitle);
-  }
-
-  void loadNextPage() {
-    if (_loadedNewsPaginatedViewEntities.hasMoreElements) {
-      _setPageLoadingState();
-      final nextPageNumber = (_loadedNewsPaginatedViewEntities.entities.length ~/ _Constants.pageSize) + 1;
-      _nyTimesPaginatedRepository
-          .getMostPopularArticlesPage(Page(nextPageNumber, _Constants.pageSize))
-          .then((articles) => _setLoadedState(articles, _loadedMostPopularViewEntity.mostPopularNewsPageTitle))
-          .catchError((error) => _onFetchError(error));
+    super.setNotifierState(const PaginatedViewState(isLoading: true));
+    _mostPopularNewsPageTitle = await _settingsRepository
+        .getPopularNewsCriterion()
+        .then((criterion) => criterion.getTitle())
+        .catchError((error) => setNotifierState(PaginatedViewState(error: error)));
+    if (super.state.error == null) {
+      super.loadFirstPage();
```

Comment:
żeby pozbyć się tego ifa na dole możnaby przypisywać wartość `_mostPopularNewsPageTitle` w środku `then` i tam też robić `loadFirstPage()` zamiast przypisywać całe wyrażenie do jednej zmiennej

Reply:
kk, chociaż przyznam, że zastanawiam się czy przypisanie do globalnej zmiennej to nie jest to side effect 🤔